jwks.go

```diff
@@ -28,10 +31,17 @@ type jsonWebKey struct {
 	ID       string `json:"kid"`
 	Modulus  string `json:"n"`
 	Type     string `json:"kty"`
+	Use      string `json:"use"`
 	X        string `json:"x"`
 	Y        string `json:"y"`
 }
 
+// parsedKey represents a parsed JWK with assoicated metadata
+type parsedKey struct {
+	use    string
+	public interface{}
+}
```

Comment:
I felt that rather than adding a second map that looks up keys by the `kid` we could instead just keep the public keys with their associated metadata. Right now it's just the `use` value, but in the future it could be useful to have other values such as `alg` (which would allow us to do smart things like support getting keys used for encryption when appropriate).

Reply:
I considered option as well and I also think it's a good way to keep track of metadata associated with the key. It'll likely be simpler and less confusing to attach more metadata in the future.

However, this will affect the `.ReadOnlyKeys()` method by changing its return type. Unfortunately, this wasn't readily apparent to someone who wasn't familiar with the code base since the return value is a `map[string]interface{}` and there was no test to confirm the `map`'s values were all within the expected set. I'd also like to add a test to this method that confirms the type of the `map` values as one of the supported cryptographic keys with a type `switch` statement.

See [the methods entry in the docs](https://pkg.go.dev/github.com/MicahParks/keyfunc#JWKS.ReadOnlyKeys).